Source/mpqapi.cpp

```diff
@@ -472,7 +461,7 @@ BOOL OpenMPQ(const char *pszArchive, BOOL hidden, int dwChar)
 	}
 	if (!sgpBlockTbl || !sgpHashTbl) {
 		memset(&fhdr, 0, sizeof(fhdr));
-		if (!ParseMPQHeader(&fhdr, &sgdwMpqOffset)) {
+		if (!ParseMPQHeader(&fhdr, (int *)&sgdwMpqOffset)) {

```

Comment:
Note, I think the signature of `ParseMPQHeader` should be `DWORD *` for the second argument, rather than `int *`. But, to keep things simple, this PR includes no changes to any other function than `mpqapi_alloc_block`.

Reply:
Since ParseMPQHeader is already bin exact and this is the only call site, it might be worth changing it now so that do don't forget.